src/users/user.service.ts

```diff
@@ -40,6 +41,59 @@ export class UserService {
     return { data: users, total };
   }
 
+  async getUserBySearch(searchUserDto: SearchUserDto): Promise<{ data: User[]; total: number }> {
+    let users = [];
+    let optionalQuery = {};
+    let splitName: string[];
+    let qFirstName, qLastName, qName, qEmail: RegExp;
```

Comment:
why define these variables in advance, except _optionalQuery_

Reply:
Corrected.